Automatically rewrite /api/2.0 to /api/1.2

JoshRosen opened the pull request:
This plugin currently does not support the `/api/2.0` endpoint and using it will result in confusing errors (as in https://github.com/databricks/sbt-databricks/issues/39).

To prevent this from confusing users, this patch updates the README to mention this limitation and adds code to automatically rewrite `/api/2.0` to `/api/1.2`. This is kind of a hack but it's a reasonable stopgap fix until we support the new API.

Tested manually.

brkyvz commented on src/main/scala/sbtdatabricks/DatabricksHttp.scala:
```diff
@@ -48,10 +48,19 @@ import sbtdatabricks.util.requests._
     outputStream: PrintStream = System.out) {
 
+  private val endpoint: String = {
+    if (_endpoint.dropRight(1).endsWith("2.")) {
```
instead of these `dropRight`s, would it be better to use split by `/` or regex matching like `/api/([1-9.]+)` to get the API version?

brkyvz commented:
Thanks for the PR @JoshRosen I left one comment

JoshRosen commented:
I finally got around to addressing feedback on this. Retested manually and my rewrite still works, so hopefully we can get this in to clear it from my open PR queue and prevent users from hitting this again.

brkyvz commented:
Thanks @JoshRosen Sorry for the delay
